realm/realm-library/src/main/cpp/io_realm_internal_OsObjectSchemaInfo.cpp

```diff
@@ -53,24 +56,33 @@ JNIEXPORT jlong JNICALL Java_io_realm_internal_OsObjectSchemaInfo_nativeGetFinal
     return reinterpret_cast<jlong>(&finalize_object_schema);
 }
 
-
-JNIEXPORT void JNICALL Java_io_realm_internal_OsObjectSchemaInfo_nativeAddProperty(JNIEnv* env, jclass,
-                                                                                   jlong native_ptr,
-                                                                                   jlong property_ptr,
-                                                                                   jboolean is_computed)
+JNIEXPORT void JNICALL Java_io_realm_internal_OsObjectSchemaInfo_nativeAddProperties(JNIEnv* env, jclass,
+                                                                                     jlong native_ptr,
+                                                                                     jlongArray j_persisted_properties,
+                                                                                     jlongArray j_computed_properties)
 {
     TR_ENTER_PTR(native_ptr)
     try {
-        ObjectSchema* object_schema = reinterpret_cast<ObjectSchema*>(native_ptr);
-        Property* property = reinterpret_cast<Property*>(property_ptr);
-        if (is_computed) {
-            object_schema->computed_properties.push_back(*property);
-        }
-        else {
-            object_schema->persisted_properties.push_back(*property);
-            if (property->is_primary) {
-                object_schema->primary_key = property->name;
+        ObjectSchema& object_schema = *reinterpret_cast<ObjectSchema*>(native_ptr);
+        JLongArrayAccessor persisted_properties(env, j_persisted_properties);
+        for (jsize i = 0; i < persisted_properties.size(); ++i)
+        {
+            Property* prop = reinterpret_cast<Property*>(persisted_properties[i]);
+            REALM_ASSERT_DEBUG(prop != nullptr);
+            if (prop->is_primary) {
+                object_schema.primary_key = prop->name;
             }
+            object_schema.persisted_properties.emplace_back(std::move(*prop));
```

Comment:
@beeender I assume the container will free the dynamically allocated pointer when destructed (since it took ownership of the pointer when using the move ctor)

Reply:
uh, no. the container will "empty" the `Property&&` which is `*prop` here. That means after this line, the `prop` is still a valid pointer which stores an empty `Property`.
The memory of the empty `Property`(pointed by `prop`) will be freed at line 76.
It will be easy to understand in this way i think:

```
Property& prop_ref = *prop; // nothing is alloced here
object_schema.persisted_properties.emplace_back(std::move(prop_ref)); 
// prop_ref has been "emptied" by the move constructor
// but it is still a valid ref since prop is still a valid pointer.
```